keyboards/macroflow_original/keymaps/default/keymap.c

```diff
@@ -0,0 +1,297 @@
+    [2] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [3] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [4] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [5] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [6] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [7] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  }
+};
+#endif
+
+#ifdef OLED_ENABLE
+
+#ifndef SHOW_LOGO
+#    define SHOW_LOGO 2500
+#endif
+static uint16_t oled_logo_timer = 0;
+static bool has_startup_ran = false;
+
+bool oled_task_user(void) {
+    if (!has_startup_ran) {
+      static const char startUp [] PROGMEM = {
+        // 'startUp', 128x32px
+        0,  0,  0,128,192,192,128,128,  0,  0,192,192,  0,  0,  0,192,192,  0,  0,128,128,192,192,128,  0,  0,192,192,192,192,192,192,192,  0,  0,128,128,192,192,128,  0,  0,  0,192,192,128,  0,  0,  0,  0,192,192,128,  0,  0,  0,128,192,192,  0,  0,  0,  0,  0,128,128,192,192,128,  0,  0,128,192,192,192,192,128,128,  0,  0,  0,128,192,192,192,128,  0,  0,  0,128,128,192,192,128,  0,  0,  0,  0,  0,  0,  0,128,192,192,128,  0,  0,  0,  0,128,192,192,128,128,  0,  0,192,192,192,  0,  0,  0,  0,128,192,192,  0,0,
+        0,  0,255,255,  1,  1,  1, 31, 30,  0,255,255,  0,  0,  0,255,255,  0, 30,127,241,193,129,  7, 15,  0,  1,  1,  1,255,255,  1,  1,  0,254,255,  1,  1,  1,255,255,  0,  0,255, 63,255,224,  0,128,254, 63,255,255,  0,  0,224,255, 15,127,255,192,  0,  0,254,255,  1,  1,  1, 31, 31,  0,255,255,129,129,129,255,127,  0,  0,255,255,  1,  1,  1,255,255,  0, 30,127,241,193,129,  7, 15,  0,  0,  0,  0,252,255,  3,  1,  1,  3, 31, 30,  0,255,255,  1,  1,  1,255,254,  0,255,255,127,252,  0,  0,240,255,255,255,  0,  0,  0,  0,255,255,128,128,192,252, 60,  0,127,255,128,128,128,255,127,  0,120,248,128,129,131,255,254,  0,  0,  0,  0,255,255,  0,  0,  0,127,255,128,128,128,255,255,  0,  0,255,  0,  3,255,248,255, 15,  0,255,255,  0,248,255, 27, 24, 24, 63,255,224,  0,127,255,128,128,128,252,124,  0,255,255,  3,  1,  3,127,254,240,  0,127,255,128,128,128,255,127,  0,120,248,128,129,131,255,254,  0,128,128,  0, 63,255,192,128,128,224,252, 60,  0,255,255,128,128,128,255,127,  0,255,255,  0, 31,255,252,127,  1,255,255,  0,  0,0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  0,  0,  1,  1,  0,  0,  0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  1,  0,  0,  0,  1,  1,  0,  0,  1,  1,  0,  1,  1,  0,  0,  0,  0,  1,  1,  0,  0,  0,  1,  1,  1,  1,  0,  0,  1,  1,  0,  0,  0,  0,  1,  1,  0,  0,  1,  1,  1,  1,  0,  0,  0,  0,  0,  1,  1,  1,  1,  0,  0,  1,  1,  0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  0,  1,  1,  1,  1,  0,  0,  0,  1,  1,  0,  0,  1,  1,  0,  0,  1,  1,  0,  0
+    };
+    oled_write_raw_P(startUp, sizeof(startUp));
+        if (timer_elapsed(oled_logo_timer) >= SHOW_LOGO) {
+            has_startup_ran = true;
+            oled_clear();
+        }
+    } else {
+  static const char PROGMEM layer[4][67+1] = {
```

Comment:
All of these could be moved to:
```suggestion
  static const char PROGMEM layer[][4][67+1] = {
```

And then you could just do:
```c
    uint8_t layer = get_highest_layer();
    oled_set_cursor(1, 0);
    oled_write_raw_P(layer[layer][0], sizeof(layer[layer][0]));
    oled_set_cursor(1, 1);  // start pos
    oled_write_raw_P(layer[layer][0], sizeof(layer[layer][0]));
    oled_set_cursor(1, 2);  // start pos
    oled_write_raw_P(layer[layer][0], sizeof(layer[layer][0]));
    oled_set_cursor(1, 3);  // start pos
    oled_write_raw_P(layer[layer][0], sizeof(layer[layer][0]));
```